src/core/ext/filters/client_channel/http_proxy.c

```diff
@@ -47,15 +50,18 @@ static char* grpc_get_http_proxy_server(grpc_exec_ctx* exec_ctx) {
     gpr_log(GPR_ERROR, "'%s' scheme not supported in proxy URI", uri->scheme);
     goto done;
   }
-  if (strchr(uri->authority, '@') != NULL) {
-    gpr_log(GPR_ERROR, "userinfo not supported in proxy URI");
-    goto done;
+  char *user_cred_end = strchr(uri->authority, '@');
+  if (user_cred_end != NULL) {
+    *name_to_resolve = gpr_strdup(user_cred_end + 1);
+    *user_cred_end = '\0';
+    *user_cred = gpr_strdup(uri->authority);
+    gpr_log(GPR_INFO, "userinfo found in proxy URI");
```

Comment:
it holds the user credentials username:password. Don't want that to show up in logs